src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/management/AnnouncementTestsIT.java

```diff
@@ -0,0 +1,159 @@
+      .body("abstract", nullValue())
+      .body("published", sameInstant(getInstant(2016, 12, 23, 11, 33, TIMEZONE_ID)));
+  } 
+  
+  @Test
+  public void testListAnnouncementsBySlug() {
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/announcements?slug=aakkostesti", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(1))
+      .body("id[0]", notNullValue())
+      .body("slug[0]", is("aakkostesti"))
+      .body("title[0]", is("Ääkköstesti"))
+      .body("abstract[0]", nullValue())
+      .body("published[0]", sameInstant(getInstant(2016, 12, 23, 11, 33, TIMEZONE_ID)));
+    
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/announcements?slug=non-existing", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(0));
+  } 
+  
+  @Test
+  public void testListAnnouncements() {
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/announcements", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(3))
+      .body("id[1]", notNullValue())
+      .body("slug[1]", is("another-test"))
+      .body("title[1]", is("Test announcement"))
+      .body("abstract[1]", nullValue())
+      .body("published[1]", sameInstant(getInstant(2016, 12, 23, 11, 33, TIMEZONE_ID)));
+  } 
+  
+  @Test
+  public void testOrganizationAnnouncementsNotFound() throws InterruptedException {
+    String organizationId = getOrganizationId(0);
+    String incorrectOrganizationId = getOrganizationId(1);
+    String organizationAnnouncecmentId = getOrganizationAnnouncementId(organizationId, 0);
+    
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    assertFound(String.format("/organizations/%s/announcements/%s", organizationId, organizationAnnouncecmentId));
+    assertEquals(3, countApiList(String.format("/organizations/%s/announcements", organizationId)));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/organizations/%s/announcements/%s", organizationId, malformedId));
+    }
+    
+    assertNotFound(String.format("/organizations/%s/announcements/%s", incorrectOrganizationId, organizationAnnouncecmentId));
+    assertEquals(0, countApiList(String.format("/organizations/%s/announcements", incorrectOrganizationId)));
+  }
+
+  
+  private void createPtvSettings() {
+    insertSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL, String.format("%s%s", getWireMockBasePath(), BASE_URL));
+    flushCache();
+  }
+  
+  private void deletePtvSettings() {
+    deleteSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL);
+  }
+    
+  private void createManagementSettings(String organizationId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "createManagementSettings" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)